fix: clearer explanation of string templates (#301)
JetBrains/kotlin-multiplatform-dev-docs: topics/compose/compose-multiplatform-resources-usage.md

* `\t` — for a tab symbol
* `\uXXXX` — for a specific Unicode character

You don't need to escape special XML characters like "@" or "?"
as you do [for Android strings](https://developer.android.com/guide/topics/resources/string-resource#escaping_quotes). 

#### String templates

Currently, arguments have basic support for string resources.

> When creating a template, use the `%<number>` format for placing arguments within the string and a `$d` or a `$s` suffix
> to signal that it's a variable placeholder and not simple text.

Reviewer: ```suggestion
When creating a template, use the `%<number>` format to place arguments within the string and include a `$d` or `$s` suffix
to indicate that it is a variable placeholder and not simple text.
```

For example:

```XML
<resources>
    <string name="str_template">Hello, %2$s! You have %1$d new messages.</string>
</resources>
```

After creating and importing the string template resource, you can refer to it while
passing the arguments for placeholders in the correct order:

```kotlin
Text(stringResource(Res.string.str_template, 100, "User_name"))
```

> There is no difference between `$s` and `$d` suffixes, and no others are supported.

Reviewer: ```suggestion
There is no difference between the `$s` and `$d` suffixes, and no others are supported.
```

> You can provide the placeholder `%1$s` and use it to display a fractional number, for example:

Reviewer: I'm unsure if "provide" is the correct verb, it feels a bit off
